lib/compiler-cranelift/src/config.rs

```diff
@@ -11,7 +11,7 @@ use wasmer_compiler::{
 /// Possible optimization levels for the Cranelift codegen backend.
 #[non_exhaustive]
 #[derive(Clone, Debug)]
-pub enum OptLevel {
+pub enum CraneliftOptLevel {
```

Comment:
Why do we need to rename this type?

Reply:
Not strictly necessary, but with this rename we avoid naming collisions in the `wasmer::` high level API namespace. In the not too far future there is probably `wasmer::OptimizationLevel` for the LLVM version, making it hard to reason about to which backends the types belong.

The other option would be use have `pub use wasmer_compiler_cranelift::{Cranelift, OptLevel as CraneliftOptLevel};`, which gives us `wasmer_compiler_cranelift::OptLevel` and `wasmer::CraneliftOptLevel`. But in my opinion this creates more confusion than it helps.

What do you think?

Reply:
Not sure about this change.
You think is likely companies will use two compilers within the same codebase? This PR moves into prefixing all the public-facing objects of compilers with `CompilernameXXX`

Reply:
> You think is likely companies will use two compilers within the same codebase?

Well, we do. I don't know much about other use cases.
But even if they don't. The caller code `use wasmer::OptLevel;` should not refer to compiler X in one project and compiler Y in a different project, right?

> This PR moves into prefixing all the public-facing objects of compilers with `CompilernameXXX`

Right. It is based on the idea that you promoted recently: make all high level API available in the `wasmer::` namespace. This is a noble goal, but hard to get right. In https://github.com/wasmerio/wasmer/pull/1730 we were fighting imports, https://github.com/wasmerio/wasmer/issues/1872 documents a conflict now we have at least very similar names for multiple compilers.

Reply:
> @webmaster128 Could you extend this renaming to other compilers in this PR, so that it's consistent across the entire codebase please?

So far this is the only new compiler-specific export in `wasmer::`:

```rust
#[cfg(feature = "singlepass")]
pub use wasmer_compiler_singlepass::Singlepass;

#[cfg(feature = "cranelift")]
pub use wasmer_compiler_cranelift::{Cranelift, CraneliftOptLevel};

#[cfg(feature = "llvm")]
pub use wasmer_compiler_llvm::LLVM;
```

However, I now also added `LLVMOptLevel` to show the direction in which this is heading.